### PR TITLE
Gem online dqm

### PR DESCRIPTION
**Pull request by hyunyong**
GEM online DQM

To be done: SQLite DB file for unpacker didn't get GT  yet

talk:
https://indico.cern.ch/event/685922/contributions/2814461/attachments/1570331/2476923/GEMDQM05DEC.pdf

@jshlee @andjuo

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

File: DQM/GEM/plugins/GEMDQMHarvester.cc [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  
+  void dqmEndLuminosityBlock(DQMStore::IBooker &, DQMStore::IGetter &, const edm::LuminosityBlock &, const edm::EventSetup &) override;
+
+  
+private:
+
+  void prova(DQMStore::IBooker &ibooker, edm::Run const &, edm::EventSetup const & iSetup);
+  std::string fName;
+  int verbosity;
+  DQMStore *dbe;
```
Comment (dmitrijus) [inline DQM/GEM/plugins/GEMDQMHarvester.cc]: please remove this pointer

File: DQM/GEM/plugins/GEMDQMStatusDigi.cc [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+    ( (TH2F *)h2GEBWarning->getTH2F() )->SetBins(10,  0, 10, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2GEBWarning->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+   
+  }
+  return mlGEBID[BID_];
+}
+
+//----------------------------------------------------------------------------------------------------
+
+void GEMDQMStatusDigi::dqmBeginRun(edm::Run const &, edm::EventSetup const &)
```
Comment (dmitrijus) [inline DQM/GEM/plugins/GEMDQMStatusDigi.cc]: please remove empty methods, you don't have to define them.

**Review by dmitrijus — CHANGES_REQUESTED** [review]

File: DQM/GEM/plugins/GEMDQMHarvester.cc (hunk 1 above)
Comment (hyunyong) [reply DQM/GEM/plugins/GEMDQMHarvester.cc]: it was done. This moment we don't use harvester, so I removed all useless things.

File: DQM/GEM/plugins/GEMDQMStatusDigi.cc (hunk 2 above)
Comment (hyunyong) [reply DQM/GEM/plugins/GEMDQMStatusDigi.cc]: it was done.

**Comment by dmitrijus** [comment]
+1

**Comment by fabiocos** [comment]
@andjuo @hyunyong @jshlee I understand that the test included in this PR has been used on real cosmics data and with GT 101X_dataRun2_HLT_Candidate_2018_03_07_17_38_52 to show that the unpacker and this DQM code works on real data. Could you please post the results of this test?

**Comment by hyunyong** [comment]
@fabiocos 
These slides show GEM online DQM and candidate GT test results.
https://indico.cern.ch/event/711282/contributions/2920924/attachments/1614003/2564012/GEMDPG08MAR.pdf

also, you can download the DQM root file at below link.
https://www.dropbox.com/s/hi1u3p8dzdn1cap/DQM_V0001_GEM_R000310292.root?dl=0

**Comment by fabiocos** [comment]
@hyunyong thanks, let's move this code forward for use at P5. This will allow to further exercise the unpacker on real data before activating it in the regular reconstruction workflows.

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
in order to use it at P5 in the short time scale we need a back-port to 10_0_X

**Pull request merged**